main.py

```diff
@@ -62,8 +62,34 @@ def h_run(args):
 		print("Failed to start: No halibot configuration found in the current directory!")
 		return
 
-	logging.basicConfig(level=logging.DEBUG)
 	bot = halibot.Halibot()
+	bot._load_config()
+
+	logfile = None
+	loglevel = logging.DEBUG
+	loglevel_str = "DEBUG"
+	loglevels = { "CRITICAL": logging.CRITICAL, "ERROR": logging.ERROR, "WARNING": logging.WARNING, "INFO": logging.INFO,
+			"DEBUG": logging.DEBUG, "NOTSET": logging.NOTSET }
+
+	if args.logfile:
+		logfile = args.logfile
+	elif "logfile" in bot.config:
+		logfile = bot.config["logfile"]
+
+	if args.loglevel:
+		loglevel = loglevels[args.loglevel]
+		loglevel_str = args.loglevel
+	elif "loglevel" in bot.config:
+		loglevel = loglevels[bot.config["loglevel"]]
+		loglevel_str = bot.config["loglevel"]
```

Comment:
I agree with this, `loglevel` assignment can be factored out to make it a bit clearer